internal/internal_worker_base.go

```diff
@@ -209,6 +211,10 @@ func (bw *baseWorker) runPoller() {
 		case <-bw.shutdownCh:
 			return
 		case <-bw.pollerRequestCh:
+			if bw.sessionTokenBucket != nil {
```

Comment:
The worker is holding a token from pollerRequestCh when doing the long poll for activity task. Also the long poll can not be cancelled, as when client side called cancel(), server side will not be notified, unless an explicit call is made. Right now the solution is using another token bucket for sessions. Before poll, poller will check if there's available token. If so, do the poll without grabbing a token from the bucket. Token is only consumed in the create session activity.